spec/index.bs

```diff
@@ -0,0 +1,404 @@
+
+  As this API relies on service workers, functionality provided by this API is only available in a <a>secure context</a>.
+</section>
+
+<section>
+  <h2 id="concepts">Concepts</h2>
+
+  The sync event is considered to run <dfn>in the background</dfn> if the user agent is either closed or no service worker clients (controlled or uncontrolled) exist for the corresponding service worker registration.
+</section>
+
+<section>
+  <h2 id="constructs">Constructs</h2>
+  A <a>service worker registration</a> has an associated <dfn>list of sync registrations</dfn> whose element type is a <a>sync registration</a>.
+
+  A <dfn>sync registration</dfn> is a tuple consisting of a <a>tag</a> and a <a lt="registration state">state</a>.
+
+  A <a>sync registration</a> has an associated <dfn>tag</dfn>, a string.
+
+  A <a>sync registration</a> has an associated <dfn>registration state</dfn>, which is one of <dfn>pending</dfn>, <dfn>firing</dfn>, <dfn>unregisteredWhileFiring</dfn>, <dfn>unregistered</dfn>, <dfn>success</dfn> or <dfn>failed</dfn>. It is initially set to <a>pending</a>.
+
+  A <a>sync registration</a> has an associated <a>service worker registration</a>. It is initially set to null.
+
+  A <a>registration state</a> is a <dfn>final registration state</dfn> if it is one of <a>unregistered</a>, <a>success</a>, or <a>failed</a>.
+
+  A <a>registration state</a> is a <dfn>firing registration state</dfn> if it is one of <a>firing</a> or <a>unregisteredWhileFiring</a>.
+
+  Within one <a>list of sync registrations</a> each <a>sync registration</a> MUST have a unique <a>tag</a>.
+</section>
+
+<section>
+  <h2 id="privacy-considerations">Privacy Considerations</h2>
+
+  <section>
+    <h3 id="location-tracking">Location Tracking</h3>
+    Fetch requests within the onsync event while <a>in the background</a> may reveal the client's IP address to the server after the user left the page. The user agent SHOULD limit tracking by capping the number of retries and duration of sync events.
+  </section>
+
+  <section>
+    <h3 id="history-leaking">History Leaking</h3>
+    Fetch requests within the onsync event while <a>in the background</a> may reveal something about the client's navigation history to passive eavesdroppers. For instance, the client might visit site https://example.com, which registers a sync event, but doesn't fire until after the user has navigated away from the page and changed networks. Passive eavesdroppers on the new network may see the fetch requests that the onsync event makes. The fetch requests are HTTPS so the request contents will not be leaked but the domain may be (via DNS lookups and IP address of the request).
+
+    Issue: Nothing is limiting fetch requests to be HTTPS only.
```

Comment:
We can update the history leaking privacy section to say that if only secure connections are allowed then only X will be leaked, if non-secure requests are allowed then Y can be leaked.

I believe in Chrome today fetches from ServiceWorkers are secure only.